### ISSUE TITLE
KeyError in asyncio cluster get_node_from_slot

### ISSUE THREAD

**Issue by sman591** [redis/redis-py#2437]
**Version**: 4.3.4

**Platform**: Python 3.8.10, AWS Elasticache, Redis 6

**Description**:

When calling `await cluster.delete(key)`, some of our machines will throw a `KeyError` (e.g. `KeyError: 14097`) from this line:

https://github.com/redis/redis-py/blob/bea00b16c8043397c6bb6f3df45d48bceefa685f/redis/asyncio/cluster.py#L1041

This isn't consistent, though. We load balance across 3 servers, and only 2 out of the 3 serve this error when it happens.

**Comment by sman591** [comment]
I think I have a reproducible example:

```
cluster = RedisCluster(
  host=config.REDIS_HOST,
  port=config.REDIS_PORT,
  decode_responses=True,
  max_connections=config.REDIS_MAX_CONNECTIONS,
  read_from_replicas=read_from_replicas,
  socket_connect_timeout=config.REDIS_CONNECT_TIMEOUT,
  skip_full_coverage_check=True,
)

await cluster.delete("foo")  # -> KeyError
```

However, if I run `cluster.ping()` first, it works:

```
cluster = RedisCluster(
  host=config.REDIS_HOST,
  port=config.REDIS_PORT,
  decode_responses=True,
  max_connections=config.REDIS_MAX_CONNECTIONS,
  read_from_replicas=read_from_replicas,
  socket_connect_timeout=config.REDIS_CONNECT_TIMEOUT,
  skip_full_coverage_check=True,
)

await cluster.ping()
await cluster.delete("foo")  # -> Success
```

**Comment by sman591** [comment]
Seems like there might be an auto-initialization piece missing in some of the asyncio cluster code, or `.initialize()` needs to be enforced as a required call

**Comment by dvora-h** [comment]
@sman591 Thanks for reporting this bug! All the details you provided helped us find exactly what is the problem.
I opened this PR to fix it: #2439

**Comment by sman591** [comment]
Sweet! That PR looks great. I'll keep an eye out for the next release.

Thank also for porting all of the asyncio + cluster support over to redis-py -- I'm migrating over two apps from aredis, and this has been invaluable
